src/backend/distributed/shared_library_init.c

```diff
@@ -195,6 +203,29 @@ _PG_init(void)
 
 
 /*
+ * multi_log_hook intercepts postgres log commands. We use this to override
+ * postgres error messages when they're not specific enough for the users.
+ */
+static void
+multi_log_hook(ErrorData *edata)
+{
+	/*
+	 * Show the user a meaningful error message when a backend is cancelled
+	 * by the distributed deadlock detection.
+	 */
+	if (edata->elevel == ERROR && edata->sqlerrcode == ERRCODE_QUERY_CANCELED &&
+		MyBackendGotCancelledDueToDeadlock())
+	{
+		edata->sqlerrcode = ERRCODE_T_R_DEADLOCK_DETECTED;
+		edata->message = "distributed deadlock detected and to resolve the deadlock "
+						 "this transaction is cancelled";
+		edata->detail = "Set citus.log_distributed_deadlock_detection to 'on' "
+						"and check server log for details.";
```

Comment:
I'm removing the detail for now. I think it makes sense that we should show this after we add more useful things to users.